### PR TITLE
Tools update

### PR DESCRIPTION
**Pull request by chris-pardy**
This Updates the build tooling significantly
* Build via `tsup`
* Linting via `eslint`
* Autoformatting with `prettier`
* Testing via `vitest`
* examples are run with `tsx`

### CODE REVIEW

File: examples/01-hello-world.mts [hunk 1]
```diff
@@ -9,50 +8,52 @@
  *   DEBUG=json-rules-engine node ./examples/01-hello-world.js
  */
 
-require('colors')
-const { Engine } = require('json-rules-engine')
+import "colors";
```
Comment (CacheControl) [inline examples/01-hello-world.mts:11]: Tell me more about the change to `.mts` decision specifically for `examples` - the idea of the examples is to cater to the lowest common denominator. Seems like we're narrowing the audience by moving away from .js?

**Review by CacheControl — COMMENTED** [review]
@chris-pardy fine with this in general.

Can we aim to get CI passing?

**Comment by chris-pardy** [comment]
> @chris-pardy fine with this in general.
> 
> Can we aim to get CI passing?

Yes, We need to drop support for node 14 & 16 which I'm going to do. I'll add an engines field to the package.json as well.

File: examples/01-hello-world.mts (hunk 1 above)
Comment (chris-pardy) [reply examples/01-hello-world.mts:11]: I guess this is more of a question of what the "lowest common denominator" is in 2024?
For instance should the examples be common js with `require` statements or can we safely move to module syntax.  Node's had support for modules for some time now and with a move to drop the EOL versions of node I think using the module syntax is probably ok.
I think for most people Typescript is the default choice in this day-and-age so it's probably worth having some examples in Typescript. I could see an argument made for actually having both.

Comment (CacheControl) [reply examples/01-hello-world.mts:11]: @chris-pardy ty, makes sense. I no longer work in this ecosystem, so I'll defer to your judgement on this.

**Pull request closed without merging**